index.js

```diff
@@ -1,20 +1,23 @@
 // taken from https://github.com/taskcluster/taskcluster-queue/blob/0dd2b45b860836a2ef6c651c5499c9abfb5670ad/utils/aws-sdk-promise.js
 var aws = require('aws-sdk');
-var Promise = require('promise');
 
 // XXX: This is a terrible hack but it should not break anything in horrible
 // ways.
 aws.Request.prototype.promise = function() {
-  return new Promise(function(accept, reject) {
-    this.on('complete', function(response) {
+  var that = this;
+  var Prom = aws.Promise || require('promise');
```

Comment:
This defaults to using a module 'promise' if the aws.Promise property doesn't exist instead of the language's Promise class.  Could we instead do 
```javascript
aws.Promise || Promise || require('promise');
```